### ISSUE TITLE
BPG+ldpc

### ISSUE THREAD

**Issue by wuchaoAAA** [KeYang8/WITT#33]
楼主，为什么我用您之前链接里面给的BPG+ldpc代码，运行后遇见的问题是如下图。楼主，您之前遇见过类似问题吗，怎么解决呀

![Image](https://github.com/user-attachments/assets/cf27dc42-1a35-4e6b-8a7d-e1b9dc457d8c)

**Comment by wuchaoAAA** [comment]
调了好久原来是错误行decoder.py中有个axis=0 改成axis=1，然后升级sonna版本

**Comment by wuchaoAAA** [comment]
还是解码不了

**Comment by DogWY** [comment]
我用链接指向的代码试了试，没碰到这个问题，感觉像是你数据集读取那的问题？

**Comment by wuchaoAAA** [comment]
Bringing hope, may I ask if you are using a 256 by 256 image  I changed the dataset and changed the 32 digits to 256在 2025年6月8日，15:32，DogWY ***@***.***> 写道：﻿DogWY left a comment (KeYang8/WITT#33)
我用链接指向的代码试了试，没碰到这个问题，感觉像是你数据集读取那的问题？

**Comment by DogWY** [comment]
我复现时使用的时CIFAR10数据集，32x32x3的没有问题，刚才试了一下将图像resize到255x255x3也没有报错

**Comment by DogWY** [comment]
由于网络原因，我无法直接使用原始的数据集逻辑，而是手动下载了CIFAR10数据集，并且通过keras的image_dataset_from_directory接口读取数据，以下附上代码以供参考：

```python
test_dataset = keras.preprocessing.image_dataset_from_directory(
    r"./data/cifar10/test",
    labels="inferred",
    label_mode="int",
    image_size=(255, 255),
    batch_size=64,
    shuffle=False,
    color_mode="rgb",
    interpolation="bilinear",
    follow_links=False,
    crop_to_aspect_ratio=False,
)
```

**Comment by wuchaoAAA** [comment]
感谢你的分享，代码调通了。请问你实现了随着snr的变化，psnr随之变化了吗

**Comment by gcding** [comment]
您好，可以提供一下BPG+LDPC的代码嘛？谢谢

1026606223@qq.com

**Comment by lovehhhhh** [comment]
> 感谢你的分享，代码调通了。请问你实现了随着snr的变化，psnr随之变化了吗

请问可以分享一下您的BPG+LDPC代码吗 1821357407@qq.com  万分感谢

**Comment by jasscia18** [comment]
你好，请问还可以分享BPG+LDPC代码吗？非常感谢    2295410401@qq.com

**Comment by dong-design** [comment]
您好打扰了，请问您有bpg+LDPC源码吗，如有可以分享给我一份吗2306683563@qq.com

> 由于网络原因，我无法直接使用原始的数据集逻辑，而是手动下载了CIFAR10数据集，并且通过keras的image_dataset_from_directory读取接口数据，以下附上代码提供参考：
> 
> test_dataset = keras.preprocessing.image_dataset_from_directory(
>     r"./data/cifar10/test",
>     labels="inferred",
>     label_mode="int",
>     image_size=(255, 255),
>     batch_size=64,
>     shuffle=False,
>     color_mode="rgb",
>     interpolation="bilinear",
>     follow_links=False,
>     crop_to_aspect_ratio=False,
> )
